usr.sbin/daemon/daemon.c

```diff
@@ -580,44 +582,44 @@ restrict_process(const char *user)
  *
  * Return value of false is assumed to mean EOF or error, and true indicates to
  * continue reading.
- *
- * TODO: simplify signature - state contains pipefd
  */
 static bool
-listen_child(int fd, struct daemon_state *state)
+listen_child(struct daemon_state *state)
 {
-	static unsigned char buf[LBUF_SIZE];
-	static size_t bytes_read = 0;
-	int rv;
+	ssize_t rv;
+	unsigned char *cp;
 
 	assert(state != NULL);
-	assert(bytes_read < LBUF_SIZE - 1);
+	assert(state->pos < LBUF_SIZE - 1);
 
-	rv = read(fd, buf + bytes_read, LBUF_SIZE - bytes_read - 1);
+	rv = read(state->pipe_fd[0], state->buf + state->pos, LBUF_SIZE - state->pos - 1);
```

Comment:
It's almost tempting to say we should, up by the `daemon_state` definition:

```
#define pipe_rd pipe_fd[0]
#define pipe_wr pipe_fd[1]
```
Then use `state->pipe_rd` and `state->pipe_wr` respectively, just to be a little neater

Reply:
I dont like marcos, instead I simply deconstructed pipe_fd[2] into 2 individual fields.